### PR TITLE
(feat) Added Model name to validate_prompt and validate_response.

### PR DESCRIPTION
**Pull request by iabouhashish**
Add Model Name Support to Inference Validation

## Summary
This PR adds support for tracking and validating model names in the GenAI Engine's inference validation system. Users can now specify which model was used for generating prompts and responses, enabling better tracking and analysis of model-specific performance.

### Core Features
- **Model Name Field**: Added `model_name` field to inference models and validation requests
- **Database Schema**: New Alembic migration adds `model_name` column to inference tables
- **API Support**: Updated validation endpoints to accept and return model names
- **Backward Compatibility**: Model name is optional, existing functionality remains unchanged

### Database Changes
- Added `model_name` column to `DatabaseInference` model
- Created Alembic migration: `2025_10_14_0905-1e4ff2a22cbb_adding_model_name.py`
- Updated repository methods to handle model name parameter

### API Updates
- **Prompt Validation**: `POST /api/v2/validate_prompt` now accepts `model_name` parameter
- **Response Validation**: `POST /api/v2/validate_response` now accepts `model_name` parameter
- **ValidationResult**: Response schema includes `model_name` field

### Testing
- **Enhanced Test Coverage**: Added comprehensive test suite for model name functionality
- **Integration Tests**: 5 new test functions with `_with_model_name` suffix
- **Test Client Updates**: Updated `GenaiEngineTestClientBase` to support model name parameter
- **Validation Assertions**: Tests verify model name is properly stored and returned

## 📊 Impact
- **Files Changed**: 9 files
- **Lines Added**: 392 lines
- **Lines Removed**: 12 lines
- **Test Coverage**: 100% coverage for new model name functionality

### CODE REVIEW

File: genai-engine/src/db_models/inference_models.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ class DatabaseInference(Base):
         lazy="joined",
     )
     task: Mapped["DatabaseTask"] = relationship(lazy="joined")
+    model_name: Mapped[str] = mapped_column(String, nullable=True)
```
Comment (alexshanabrook) [inline genai-engine/src/db_models/inference_models.py]: Curious, why did you did you decide to put this here instead of in `DatabaseInferenceResponse` or `DatabaseInferencePrompt`? I don't know enough about the GenAI resources to be sure which would be expected, so I'm not saying it needs to be moved, just wanted a little more context

File: genai-engine/src/validation/prompt.py [hunk 2]
```diff
@@ -27,10 +28,12 @@ def validate_prompt(
...
         conversation_id=body.conversation_id,
         user_id=body.user_id,
+        model_name=body.model_name,
     )
 
     return ValidationResult(
         inference_id=inference_prompt.inference_id,
         rule_results=inference_prompt._to_response_model().prompt_rule_results,
         user_id=body.user_id,
+        model_name=body.model_name,
```
Comment (alexshanabrook) [inline genai-engine/src/validation/prompt.py]: also curious how this is relevant to a `ValidationResult`? It seems like it's more being handled as new metadata for a `PromptValidationRequest`, in which case it's not really a result of the validation so much as a config on the initial inference?

File: genai-engine/tests/integration/test_rule_execution.py [hunk 3]
```diff
@@ -313,3 +314,322 @@ def test_rule_result_skipped(client: GenaiEngineTestClientBase):
...
+
+
+@pytest.mark.parametrize(
+    ("prompt_str", "expected_rule_results"),
+    [
+        ["Hello World", [RuleResultEnum.PASS]],
+        [" ".join(["Hello World" for _ in range(5000)]), [RuleResultEnum.SKIPPED]],
+    ],
+)
+@pytest.mark.integration_tests
```
Comment (alexshanabrook) [inline genai-engine/tests/integration/test_rule_execution.py]: could you help me understand how this case & all of the ones below it are edge cases for the handling of `model_name`? To me it seems like they're testing non-model name code paths (as in, cases that have nothing to do with whether or not `model_name` is being set correctly & will have the same behavior every time). If that's the case, could we remove them? I want to be careful of adding unnecessary tests that we'll need to maintain moving forward if they're not testing specific cases we're interested in

Comment (iabouhashish) [reply genai-engine/tests/integration/test_rule_execution.py]: It's just to make sure that adding model_name hasn't resulted in any changes to the tests above, which don't contain model_name. I can remove most of them if we think this is overkill.

File: genai-engine/src/schemas/internal_schemas.py [hunk 4]
```diff
@@ -1166,6 +1170,7 @@ class Inference(BaseModel):
     inference_response: Optional[InferenceResponse] = None
     inference_feedback: List[InferenceFeedback]
     user_id: Optional[str] = None
+    model_name: Optional[str] = None
```
[the remark this answers is not in the document]
Comment (iabouhashish) [reply genai-engine/src/schemas/internal_schemas.py]: Same as above. I don't foresee model_name being used any differently currently than user_id, so following the same pattern.

**Comment by iabouhashish** [comment]
> > @videetparekh @alexshanabrook I have now removed it from validate_prompt, only added to validate_response.
> 
> Sorry, how did you do that? Don't you have to remove it from arthur-common first here? https://github.com/arthur-ai/arthur-common/blob/bf1950ab2e847fbff5525fc69cacd0725442cc15/src/arthur_common/models/request_schemas.py#L334

Here: https://github.com/arthur-ai/arthur-common/pull/56
